### PR TITLE
refactor: remove configurator in scheduler

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup
/sig scheduling

#### Which issue(s) this PR fixes:

Part 1 of https://github.com/kubernetes/kubernetes/issues/108788

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -305,45 +307,78 @@ func New(client clientset.Interface,
...
+	clusterEventMap := make(map[framework.ClusterEvent]sets.String)
 
-	// Create the config from component config
-	sched, err := configurator.create()
+	profiles, err := profile.NewMap(options.profiles, registry, recorderFactory,
+		frameworkruntime.WithComponentConfigVersion(options.componentConfigVersion),
+		frameworkruntime.WithClientSet(client),
+		frameworkruntime.WithKubeConfig(options.kubeConfig),
+		frameworkruntime.WithInformerFactory(informerFactory),
+		frameworkruntime.WithSnapshotSharedLister(snapshot),
```
Comment (kerthcet) [inline pkg/scheduler/scheduler.go:333]: remove `frameworkruntime.WithRunAllFilters(c.alwaysCheckAllPredicates)` here as `c.alwaysCheckAllPredicates` is always false now. I also create a issue https://github.com/kubernetes/kubernetes/issues/108770 to discuss about whether we should remove this field.

File: pkg/scheduler/factory.go [hunk 2]
```diff
@@ -18,182 +18,81 @@ package scheduler
...
-		internalqueue.MakeNextPodFunc(podQueue),
-		MakeDefaultErrorFunc(c.client, c.informerFactory.Core().V1().Pods().Lister(), podQueue, c.schedulerCache),
-		c.StopEverything,
-		podQueue,
-		profiles,
-		c.client,
-		c.nodeInfoSnapshot,
-		c.percentageOfNodesToScore)
-	return sched, nil
+	return
```
Comment (ahg-g) [inline pkg/scheduler/factory.go:95]: remove this

File: pkg/scheduler/scheduler.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/scheduler.go:333]: yes, we can remove it

File: pkg/scheduler/factory.go [hunk 3]
```diff
@@ -18,182 +18,81 @@ package scheduler
...
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/noderesources"
-	frameworkruntime "k8s.io/kubernetes/pkg/scheduler/framework/runtime"
 	internalcache "k8s.io/kubernetes/pkg/scheduler/internal/cache"
-	cachedebugger "k8s.io/kubernetes/pkg/scheduler/internal/cache/debugger"
 	internalqueue "k8s.io/kubernetes/pkg/scheduler/internal/queue"
-	"k8s.io/kubernetes/pkg/scheduler/profile"
 )
 
 // Binder knows how to write a binding.
 type Binder interface {
```
Comment (ahg-g) [inline pkg/scheduler/factory.go:37]: can we remove this? is is actually used anywhere?

File: pkg/scheduler/scheduler.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/scheduler.go:333]: can you cleanup `WithRunAllFilters` and `runAllFilters` from the framework too

Comment (kerthcet) [reply pkg/scheduler/scheduler.go:333]: This is still be discussed in issue #108770 , if we come to an agreement, I'll create another pr to follow this.

File: pkg/scheduler/factory.go (hunk 3 above)
Comment (kerthcet) [reply pkg/scheduler/factory.go:37]: Seems unused anywhere. I'll remove this. Thanks.

File: pkg/scheduler/factory.go (hunk 2 above)
Comment (kerthcet) [reply pkg/scheduler/factory.go:95]: Personally would like to keep it as more clearly to me.

Comment (ahg-g) [reply pkg/scheduler/factory.go:90]: what is the confusion that you try to clear? it is not typical to explicitly return at the end of a function that returns nothing.

Comment (ahg-g) [reply pkg/scheduler/factory.go:90]: ah, you have named return variable. So the best practice is not to use named variables on long functions, can you please explicitly define them in the function and in the return statement.

Comment (kerthcet) [reply pkg/scheduler/factory.go]: I used to follow the practice: `The names are not mandatory but they can make code shorter and clearer: they're documentation` in https://go.dev/doc/effective_go#named-results. 
Updated then.

File: pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -305,45 +307,78 @@ func New(client clientset.Interface,
...
+		schedulerCache,
+		podQueue,
+	)
+	debugger.ListenForSignal(stopEverything)
+
+	sched := newScheduler(
+		schedulerCache,
+		extenders,
+		internalqueue.MakeNextPodFunc(podQueue),
+		MakeDefaultErrorFunc(client, informerFactory.Core().V1().Pods().Lister(), podQueue, schedulerCache),
```
Comment (denkensk) [inline pkg/scheduler/scheduler.go:373]: We can define a variable for `informerFactory.Core().V1().Pods().Lister()` as it's long and used several times in this function after this reactor. :)

Comment (kerthcet) [reply pkg/scheduler/scheduler.go:373]: Marked it down. I'll do this in another follow-up PR. Thanks.

Comment (denkensk) [reply pkg/scheduler/scheduler.go:373]: Not need to do it in other pr or commit. This is introduced by this refactoring.

Comment (kerthcet) [reply pkg/scheduler/scheduler.go:373]: I haven't checked it yet. Let me see.

Comment (kerthcet) [reply pkg/scheduler/scheduler.go:373]: done :)

**Comment by ahg-g** [comment]
thanks, can you please squash.

**Comment by kerthcet** [comment]
squashed.

**Comment by ahg-g** [comment]
another followup is to re-organize the code as follows:
1) remove factory.go file and move the remaining functions to scheduler.go.
2) add a new file named `schedule_one.go`, which would include the `scheduleOne` func and all other local functions it invokes.

With this, scheduler.go would be the file the includes the instantiation logic, while schedule_one.go includes the scheduling logic.

**Pull request merged**